src/components/calcite-picker-item/calcite-picker-item.tsx

```diff
@@ -0,0 +1,180 @@
+
+  @Prop({ reflect: true }) value: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  @State() isSelected = this.selected;
+
+  dir: "rtl" | "ltr";
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    this.dir = getElementDir(this.el);
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calcitePickerItemSelectedChange: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async toggleSelected(coerce?: boolean, emit = false) {
+    this.isSelected = typeof coerce === "boolean" ? coerce : !this.isSelected;
+    if (emit) {
+      this.emitChangeEvent();
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  pickerClickHandler = (event: MouseEvent): void => {
+    this.isSelected = !this.isSelected;
+    this.emitChangeEvent(event.shiftKey);
+  };
+
+  secondaryActionContainerClickHandler(event: MouseEvent) {
+    event.stopPropagation();
+  }
+
+  emitChangeEvent(shiftPressed = false) {
+    this.calcitePickerItemSelectedChange.emit({
+      item: this.el,
+      value: this.value,
+      selected: this.isSelected,
+      shiftPressed
+    });
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  renderIcon() {
+    const { icon } = this;
+    if (!icon) {
+      return null;
+    }
+    if (icon === ICON_TYPES.grip) {
+      return (
+        <span class={CSS.handle}>
+          <CalciteIcon size="24" path={handleVertical24} />
+        </span>
+      );
+    } else {
+      const path =
+        icon === ICON_TYPES.square
+          ? this.isSelected
+            ? checkSquare16
+            : square16
+          : this.isSelected
+          ? circleFilled16F
+          : circle16F;
+      return (
+        <span class="icon">
+          <CalciteIcon size="16" path={path} />
+        </span>
+      );
+    }
+  }
+
+  render() {
+    const { icon } = this;
+    return (
+      <Host
+        dir={this.dir}
+        class={icon !== ICON_TYPES.square && icon !== ICON_TYPES.circle ? CSS.highlight : null}
```

Comment:
Should we be setting classes on the host element? @jcfranco

Reply:
IMO, I don't think we should be setting classes on the host element. These should probably be in the shadowDOM.

Reply:
I remembered reading that somewhere.
https://developers.google.com/web/fundamentals/web-components/best-practices#do-not-self-apply-classes

The highlight will go away once we split these - so that will work itself out. 

But the dir I changed to use a class per other feedback - so we can go back to using an attribute for it. I'd like to avoid adding another wrapping element if its redundant. We could use something other than dir if there's concern that will subvert changes to the parent tree, but even then there will be weirdness if this component detects a dir on load. Let's chat about how best to handle this.